docs/update-from-client-side-helpers-to-server-side.md

```diff
@@ -0,0 +1,107 @@
+            // Finalize the transaction
+            onApprove: function (data, actions) {
+-             return actions.order
+-               .capture()
+-               .then(function (orderData) {
+-                  // Successful capture!
+-                })
+-                .catch(function (err) {
+-                   // Failed capture
+-                });
++                return fetch("/my-server/patch-paypal-order", {
++                  method: "POST",
++                })
++                .then((response) => response.json())
++                .then(function (orderData) {
++                   // Successful capture!
++                })
++                .catch(function (err) {
++                   // Failed capture
++                });
+            },
+            onError: function (err) {
+              // For example, redirect to a specific error page
+              window.location.href = "/your-error-page-here";
+            }
+        })
+        .render("#paypal-button-container");
+    </script>
+  </body>
+</html>
+```
+
+- What is server-side code?
+
+  Server-side code runs securely on a web server and is typically used to communicate with APIs and Databases. Here's a list of common server-side languages used to make websites: Node.js, PHP, ASP.NET, Ruby, Java.
+
+- What should I do if I do not have the ability to run server-side code?
+
+  We recommend using one of PayPal's partners to host your website like Wix, GoDaddy, Shopify, and Big Commerce.
+
+## Best Practices for JS SDK Server-Side Integrations:
+
+1. CLIENT_SECRET should never be checked into git. We recommend passing this sensitive value to the web server at runtime as an environment variable. It's common to use a .env file that is ignored by git to load sensitive values like this secret.
+Examples:
+  * Node.js: `PAYPAL_CLIENT_SECRET=ABC_123_DOE_RAY_ME node server.js` and use `process.env` to read this value.
+  * Python: `PAYPAL_CLIENT_SECRET=ABC_123_DOE_RAY_ME my_cool_application.py` and use `os.environ` to read this value.
+
+2. The authorization token returned by `/v1/oauth2/token` api endpoint should never be passed to the browser. Keep this value in memory on the server-side and use it as the Authorization header for all other api calls.
+
+3. Create API endpoints to wrap the PayPal APIs. These API endpoints should include error handling.
+  * for errors, reply with the response body since it contains helpful information about the error (ex: 'INSTRUMENT_DECLINED' error when capturing an order)
+  * for errors, reply with the http status code since it provides value for troubleshooting purposes (ex: 4xx errors for bad user input vs 5xx errors for internal failures)
+    
+4. We recommend hosting your API endpoints on the same domain as your website. This way you do not have to setup CORS for the browser fetch() calls. If you need to use a different domain/sub-domain, we recommend setting up CORS to only be able to talk to that domain and no others.
```

Comment:
I'm thinking we remove this point about CORS. It's probably in the weeds too much.
```suggestion
```